flopy/utils/binaryfile.py

```diff
@@ -848,6 +850,8 @@ def _skip_record(self, header):
             # read rest of list data
             nauxp1 = binaryread(self.file, np.int32)[0]
             naux = nauxp1 - 1
+            if naux > 100:

```

Comment:
Hey @jlarsen-usgs, is this the only way to get this to work by hardwiring in 100?

Reply:
@langevin-usgs 

nauxp1 is read as a very large number when the precision is incorrect. I can look to see if there are other methods to check for this.

Reply:
![image](https://user-images.githubusercontent.com/3372403/81722857-9fa1ee80-9447-11ea-906d-a9a02679fec4.png)
If this is the structure of imeth=6, for example, shouldn't we know if there is failure if text doesn't contain ascii characters?  nauxp1 has garbage in it if the precision is wrong because delt/pertim/totim won't be read correctly.  We've tried to do this try/except before but it wasn't reliable so we put checks in (like nauxp < 100) and invariably we found someone with a valid case that was flagged as a problem (e.g. someone will have more than 100 auxiliary variables soon!).

Reply:
@langevin-usgs 

That's actually what I'm looking at now. There is garbage in the 'modelnam' field of the header if we're using the wrong precision. After the tests finish running on my local machine I'll update this PR.

Reply:
@jlarsen-usgs I can think of one method to check for this without hard wiring numbers:

1) Use naux to calculate where nlist should be in the file, if nlist is past the end of the file report an error, otherwise go to 2)
2) Seek to the nlist location in the file and read nlist. Using nlist and naux calculate nbytes. If nbytes + the current file offset is greater than the end of file, report an error.

Also, report an error if naux or nlist are less than or equal to 0.

Assuming naux is very large when the precision is incorrect, it will get to the end of the file before it finishes building the indexes and will report an error.  To be safe you can also check in _build_index that ipos of the last record plus the length of the last record is equal to the file size, since I suspect it is possible that naux could be very small in some rare circumstances.

Reply:
Updated to check an additional header field for binary junk